src/bufr/BufrParser/BufrParser.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * (C) Copyright 2020 NOAA/NWS/NCEP/EMC
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "BufrParser.h"
+
+#include <iostream>
+#include <map>
+
+#include "bufr.interface.h"
+
+#include "BufrParser/BufrCollectors/BufrCollectors.h"
+#include "BufrMnemonicSet.h"
+#include "DataContainer.h"
+
+#include "Exports/MnemonicExport.h"
+#include "Exports/DatetimeExport.h"
+#include "Exports/Export.h"
+
+
+namespace Ingester
+{
+    static const unsigned int SUBSET_STR_LEN = 25;
```

Comment:
Move into BufrParser::parse.

Reply:
Also, why is the length 25?

Reply:
The Fortran call to ireadmsg (NCEPLIBS-bufr) requires a pre-allocated char array to hold the subset string for the message being parsed. I just picked 25 because its long enough to hold any subset string (it was kinda an arbitrary choice).